app/models/pet.rb

```diff
@@ -19,4 +13,12 @@ def applicant
     pet_applications.where(adopted: true).first.application
   end
 
+  def self.pets_with_applications
+    select(:name, :id).joins(:pet_applications)
+  end
+
+  def self.fav_pets(favorites)
+    find(favorites.pets_favorite.keys)
+  end
```

Comment:
This basically just refactors the old method to avoid using map and makes it one line.

Reply:
I like that. I think that eliminates ruby methods, correct?